### PR TITLE
Add CIS with CUHF reference example notebook.

### PR DESCRIPTION
**Pull request by dariavh**
**Short description**

This notebook can be used for @ruvdrsti's reference CIS calculations with a CUHF determinant.

**To do**

- [x] (if applicable, when creating new source files): don't forget to update the Doxygen file, the collective `gqcp.hpp` include header and the CMake files
- [x] Add other smaller task to be completed as a Markdown task list

### CODE REVIEW

**Comment by lelemmen** [comment]
@dariatols Could you add the Jupyter Notebook to the website's examples section?

**Comment by dariatols** [comment]
@lelemmen Sure, I'll do this next week!

**Comment by lelemmen** [comment]
This PR seems a little stale. Are there things that still need to happen?

**Comment by dariatols** [comment]
@lelemmen My recent commit has taken care of cleaning up this notebook.

File: gqcp/tests/data/h3.xyz [hunk 1]
```diff
@@ -0,0 +1,5 @@
+3
```
Comment (lelemmen) [inline gqcp/tests/data/h3.xyz:1]: I don't think you're using this file anywhere. Could you delete it?

File: website/examples/CIS with CUHF reference.ipynb [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+   "source": [
+    "## Molecular setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "outstanding-change",
+   "metadata": {},
+   "source": [
+    "Let us start by defining the molecular system under consideration. We'll take an H3-ring, 1 a.u. apart and in the STO-3G basisset."
```
Comment (lelemmen) [inline website/examples/CIS with CUHF reference.ipynb:35]:
```suggestion
    "Let us start by defining the molecular system under consideration. We'll take an H3-ring, 1 angstrom apart and in the STO-3G basisset."
```

File: website/examples/CIS with CUHF reference.ipynb [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+    "print(\"Number of alpha electrons: {}\".format(N_alpha))\n",
+    "print(\"Number of beta electrons: {}\".format(N_beta))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "expanded-practitioner",
+   "metadata": {},
+   "source": [
+    "In order to start any calculations, we need to calculate the molecular integrals. In `GQCP`, this is done by creating a restricted spin-orbital basis:"
```
Comment (lelemmen) [inline website/examples/CIS with CUHF reference.ipynb:71]:
```suggestion
    "In order to start any unrestricted calculations, we need to calculate the molecular integrals. In `GQCP`, this is done by creating an unrestricted spin-orbital basis:"
```

**Review by lelemmen — APPROVED** [review]

**Pull request merged**